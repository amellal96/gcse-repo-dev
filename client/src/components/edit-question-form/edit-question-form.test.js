import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk'
import {Provider} from 'react-redux';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';

import EditQuestionForm from './edit-question-form.component';

const mockStore = configureStore([thunk]);

describe('EditQuestionForm component', () => {
    let store;
    let wrapper;

    beforeEach(() => {
        store = mockStore({
            question: {
                question: {

                }
            }
        });

        store.dispatch = jest.fn();

        const mockProps = {
            store
        }

        wrapper = mount(
            <Router>
                <Provider store={store}>
                    <EditQuestionForm {...mockProps} />
                </Provider>
            </Router>
        );
    });

    it('should match the snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot(); 
    });

    it('should render form if question found', () => {
        expect(wrapper.find('.upload-question')).toHaveLength(1);
    });
});