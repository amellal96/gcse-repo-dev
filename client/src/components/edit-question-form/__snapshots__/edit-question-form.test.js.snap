// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditQuestionForm component should match the snapshot 1`] = `"<div class=\\"upload-question\\"><h2>Edit Question</h2><form class=\\"question-container\\"><div class=\\"question-answer\\"><h3>Question</h3><label for=\\"exampleTextarea\\">Add Indices</label><div class=\\"btn-toolbar\\" role=\\"toolbar\\" aria-label=\\"Toolbar with button groups\\"><div class=\\"btn-group mr-2\\" role=\\"group\\" aria-label=\\"First group\\"><button type=\\"button\\" value=\\"²\\" class=\\"btn btn-secondary\\">²</button><button type=\\"button\\" value=\\"³\\" class=\\"btn btn-secondary\\">³</button><button type=\\"button\\" value=\\"⁴\\" class=\\"btn btn-secondary\\">⁴</button></div><div class=\\"btn-group mr-2\\" role=\\"group\\" aria-label=\\"Second group\\"><button type=\\"button\\" value=\\"°\\" class=\\"btn btn-secondary\\">°</button></div></div><textarea class=\\"form-control\\" rows=\\"3\\" type=\\"text\\" name=\\"questionText\\" label=\\"Question\\"></textarea><h3>Answer</h3><label for=\\"exampleTextarea\\">Add Indices</label><div class=\\"btn-toolbar\\" role=\\"toolbar\\" aria-label=\\"Toolbar with button groups\\"><div class=\\"btn-group mr-2\\" role=\\"group\\" aria-label=\\"First group\\"><button type=\\"button\\" value=\\"²\\" class=\\"btn btn-secondary\\">²</button><button type=\\"button\\" value=\\"³\\" class=\\"btn btn-secondary\\">³</button><button type=\\"button\\" value=\\"⁴\\" class=\\"btn btn-secondary\\">⁴</button></div><div class=\\"btn-group mr-2\\" role=\\"group\\" aria-label=\\"Second group\\"><button type=\\"button\\" value=\\"°\\" class=\\"btn btn-secondary\\">°</button></div></div><textarea class=\\"form-control\\" rows=\\"3\\" type=\\"text\\" name=\\"answer\\" label=\\"Answer\\"></textarea></div><div class=\\"question-properties\\"><h3>Exam Boards</h3><div class=\\"option-selection\\" value=\\"\\"><span class=\\"question-option-container\\"><input class=\\"question-option\\" type=\\"checkbox\\" name=\\"AQA\\" value=\\"AQA\\"> AQA</span><span class=\\"question-option-container\\"><input class=\\"question-option\\" type=\\"checkbox\\" name=\\"Edexcel\\" value=\\"Edexcel\\"> Edexcel</span><span class=\\"question-option-container\\"><input class=\\"question-option\\" type=\\"checkbox\\" name=\\"OCR\\" value=\\"OCR\\"> OCR</span><span class=\\"question-option-container\\"><input class=\\"question-option\\" type=\\"checkbox\\" name=\\"WJEC\\" value=\\"WJEC\\"> WJEC</span></div><h3>Topics</h3><div class=\\"option-selection\\" value=\\"\\"><span class=\\"question-option-container\\"><input class=\\"question-option\\" type=\\"checkbox\\" name=\\"Number\\" value=\\"Number\\"> Number</span><span class=\\"question-option-container\\"><input class=\\"question-option\\" type=\\"checkbox\\" name=\\"Algebra\\" value=\\"Algebra\\"> Algebra</span><span class=\\"question-option-container\\"><input class=\\"question-option\\" type=\\"checkbox\\" name=\\"Ratio, Proportion and Rates of change\\" value=\\"Ratio, Proportion and Rates of change\\"> Ratio, Proportion and Rates of change</span><span class=\\"question-option-container\\"><input class=\\"question-option\\" type=\\"checkbox\\" name=\\"Geometry and Measures\\" value=\\"Geometry and Measures\\"> Geometry and Measures</span><span class=\\"question-option-container\\"><input class=\\"question-option\\" type=\\"checkbox\\" name=\\"Statistics &amp; Probability\\" value=\\"Statistics &amp; Probability\\"> Statistics &amp; Probability</span></div><h3>Total Marks</h3><div class=\\"group\\"><label class=\\" form-input-label\\">Marks</label><input class=\\"form-control\\" type=\\"text\\" name=\\"marks\\" value=\\"\\"></div><h3>Grade</h3><select class=\\"form-control small\\" name=\\"difficulty\\"><option>1</option><option>2</option><option>3</option><option>4</option><option>5</option><option>6</option><option>7</option><option>8</option><option>9</option></select></div><input type=\\"submit\\" class=\\"btn btn-primary btn-lg\\" value=\\"UPDATE\\"></form></div>"`;
